File: CSharpCore/HamaraBasketQualityAssurance/MovieTicketsQualityChecker.cs
namespace HamaraBasketQualityAssurance
{
    public class MovieTicketsQualityChecker : QualityChecker
    {
       
        public override void UpdateQuality(HamaraBasketItem item)
        {
            if(item.SellBy<=0)
            {
                item.Quality = 0;
                return;
            }
            if(item.SellBy<=5)
            {
                item.Quality += 3;
                return;
            }
            if(item.SellBy<=10)
            {
                item.Quality += 2;
                return;
            }
            item.Quality += 1;
        }
    }
}

File: Java/src/main/java/com/hamaraBasket/MovieTicketsQualityChecker.java
package com.hamaraBasket;

public class MovieTicketsQualityChecker extends QualityChecker
    {
       @Override
        public void updateQuality(HamaraBasketItem item)
        {
            if(item.getSellBy()<=0)
            {
                item.setQuality(0);
                return;
            }
            if(item.getSellBy()<=5)
            {
                int quality = item.getQuality();
                item.setQuality(quality+3);
                return;
            }
            if(item.getSellBy()<=10)
            {
                int quality = item.getQuality();
                item.setQuality(quality+ 2);
                return;
            }

            int quality = item.getQuality();
            item.setQuality(quality+ 1);
        }
    }